storage.js

```diff
@@ -0,0 +1,89 @@
+const funcs = [];
+funcs.push(require('./models/users.js'));
+funcs.push(require('./models/challenges.js'));
+funcs.push(require('./models/roles.js'));
+funcs.push(require('./models/user_roles.js'));
+funcs.push(require('./models/role_permissions.js'));
+
+const Schema = require('caminte').Schema
+
+memoryUpdate = function (model, filter, data, callback) {
+  'use strict';
+  if ('function' === typeof filter) {
+    return filter(new Error('Get parametrs undefined'), null);
+  }
+  if ('function' === typeof data) {
+    return data(new Error('Set parametrs undefined'), null);
+  }
+  filter = filter.where ? filter.where : filter;
+  var mem = this;
+
+  // filter input to make sure it only contains valid fields
+  var cleanData = this.toDatabase(model, data);
+
+  if (data.id) {
+    // should find one and only one
+    this.exists(model, data.id, function (err, exists) {
+      if (exists) {
+        mem.save(model, Object.assign(exists, cleanData), callback);
+      } else {
+        callback(err, cleanData);
+      }
+    })
+  } else {
+    this.all(model, filter, function(err, nodes) {
+      var count = nodes.length;
+      if (!count) {
+        return callback(false, cleanData);
+      }
+      nodes.forEach(function(node) {
+        mem.cache[model][node.id] = Object.assign(node, cleanData);
+        if (--count === 0) {
+          callback(false, cleanData);
+        }
```

Comment:
Not sure I understand here, looks like this is just meant to trigger after the forEach has executed on every element?
In which case can just put the callback call after the forEach and remove the count?